File: solutions/280.wiggle-sort.java
class Solution {
    public void wiggleSort(int[] nums) {
        for(int i=0; i+1<nums.length; ++i) {
            if ((i%2==0 && nums[i]>nums[i+1]) || (i%2==1 && nums[i]<nums[i+1])) {
                swap(nums, i, i+1);
            }
        }
    }

    void swap(int[] nums, int i, int j) {
        var temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: solutions/280.wiggle-sort.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var wiggleSort = function (nums) {
  const swap = (i, j) => {
    [nums[i], nums[j]] = [nums[j], nums[i]];
  };

  for (let i = 0; i < nums.length - 1; ++i) {
    if (i % 2 === 0) {
      if (nums[i] > nums[i + 1]) swap(i, i + 1);
    } else {
      if (nums[i] < nums[i + 1]) swap(i, i + 1);
    }
  }
};
